### ISSUE TITLE
WHEN_WITHOUT_ELSE doesn't recognize when expression in when branch

### ISSUE THREAD

**Issue by petertrr** [cqfn/diKTat#415]
## Describe the bug
```kotlin
when (expr1) {
    is Type1 -> when (expr1.field) {
                is Type2 -> when (foo(epxr1)) {
                    SomeEnum.Field1 -> bar()
                    SomeEnum.Field2 -> baz()
                }
}
```
the `when` in `is Type2` is not treated as an expression (it's treated as a statement, though it's an expression).

## Expected behavior
WHEN_WITHOUT_ELSE shouldn't check when expressions at all until we have type resolution.

## Environment information
* diktat version: 0.1.2

**Comment by petertrr** [comment]
Another false-positive: when expression is assigned to a var:
`var x: Int
x = when () ...
`
